### DISCUSSION TITLE
How to get registered route?

### DISCUSSION THREAD

**Discussion by Dzyszla** [HashLoad/horse#402] [Q&A]
I have registered some function, for example:
```delphi
procedure MyFunction(req: THorseRequest, resp: THorseResponse);
var
  i: Integer;
begin
   i := MyDatas.IndexOf(....?); // Req.InfoPath contains real called path, not registered path
end;

...

var
   MyRoutes: TStrings;
   s: String;
begin
   ...
  MyRoutes.Add('/fn');
  MyRoutes.Add('/fn/:Param1');
  MyRoutes.Add('/fn/:param1/:param2');

  for s in MyRoutes do
  begin
     THorse.Get(s, MyFunction);
     MyDatas.Add(s);
  end;
end;
```
Is there a way to extract the routing used in a function? I can't extract the parameter names directly from the `Request` property because there's a `TDictionary` there that doesn't return them in the order I asked for.

**Comment by jcaique** [comment]
PathInfo property inside THorseRequest doesnt return this information?

**Reply by Dzyszla** [reply]
Nope. If I type 'localhost/fn/2/2', then in `PathInfo` is '/fn/2/2', not '/fn/:param1/:param2'.
